staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -23,8 +23,10 @@ import (
 	"k8s.io/utils/clock"
 )
 
+// Interface defines the minimal requirements to implement a workqueue.
 type Interface interface {
 	Add(item interface{})
+	IsQueued(item interface{}) bool
```

Comment:
Do we have a clear usecase for it?

Changing the interface means that if someone was implementing the queue downstream, updating the client-go dependency will break them. I would like to ensure that we have a good reason for doing that.

Reply:
https://github.com/kubernetes/kubernetes/pull/112328 would be one use case for it (in order to offer better queuing options we need to be able to inspect the queue membership).

Any client of the queue that wants to be able to make a decision to queue based on whether the item is already queued needs a way of interrogating the queue.  I note that we already implement Len() which allows a client to inspect the length of the queue, and I think IsQueued() is similarly useful.

I don't believe, in this case, that it would be very hard for a client to add this method their custom queue implementation or mocks.

Reply:
to add more details, this is one of the issues the PR referenced above will eventually fix: https://github.com/kubernetes/kubernetes/issues/110642